src/main/java/com/blamejared/recipestages/handlers/Recipes.java

```diff
@@ -95,10 +97,21 @@ public static void addShapeless(String stage, IItemStack output, IIngredient[] i
     
     @ZenMethod
     public static void setRecipeStage(String stage, IIngredient output) {
-        List<IRecipe> recipes = new ArrayList<>();
-        ForgeRegistries.RECIPES.getEntries().stream().filter(recipe -> output.matches(CraftTweakerMC.getIItemStack(recipe.getValue().getRecipeOutput()))).forEach(recipe -> {
-            recipes.add(recipe.getValue());
-        });
+        if (recipeOutputCache.isEmpty()) {
```

Comment:
This is way to early to create a cache, not all mods would have registered their recipes, so what *should* happen is that the ActionSetStage should be changed to accept a (String stage, IIngredient output), and this logic be done in the action (The action is delayed until just before JEI runs, most actions are, it ensures that all the recipes are there)